openedx/features/enterprise_support/api.py

```diff
@@ -308,34 +303,38 @@ def enterprise_customer_for_request(request):
             # Check if there's an Enterprise Customer such that the linked SSO provider
             # has an ID equal to the ID we got from the running pipeline or from the
             # request tpa_hint URL parameter.
-            ec_uuid = EnterpriseCustomer.objects.get(
+            enterprise_customer_uuid = EnterpriseCustomer.objects.get(
                 enterprise_customer_identity_provider__provider_id=sso_provider_id
             ).uuid
         except EnterpriseCustomer.DoesNotExist:
             # If there is not an EnterpriseCustomer linked to this SSO provider, set
             # the UUID variable to be null.
-            ec_uuid = None
+            enterprise_customer_uuid = None
     else:
         # Check if we got an Enterprise UUID passed directly as either a query
         # parameter, or as a value in the Enterprise cookie.
-        ec_uuid = request.GET.get('enterprise_customer') or request.COOKIES.get(settings.ENTERPRISE_CUSTOMER_COOKIE_NAME)
+        enterprise_customer_uuid = request.GET.get('enterprise_customer') or request.COOKIES.get(
+            settings.ENTERPRISE_CUSTOMER_COOKIE_NAME
+        )
 
-    if not ec_uuid and request.user.is_authenticated():
+    if not enterprise_customer_uuid and request.user.is_authenticated():
         # If there's no way to get an Enterprise UUID for the request, check to see
         # if there's already an Enterprise attached to the requesting user on the backend.
         learner_data = get_enterprise_learner_data(request.site, request.user)
         if learner_data:
-            ec_uuid = learner_data[0]['enterprise_customer']['uuid']
-    if ec_uuid:
+            enterprise_customer_uuid = learner_data[0]['enterprise_customer']['uuid']
+    if enterprise_customer_uuid:
         # If we were able to obtain an EnterpriseCustomer UUID, go ahead
         # and use it to attempt to retrieve EnterpriseCustomer details
         # from the EnterpriseCustomer API.
         try:
-            ec = EnterpriseApiClient().get_enterprise_customer(ec_uuid)
+            enterprise_customer = EnterpriseApiClient(user=request.user).get_enterprise_customer(
```

Comment:
I think you will need to use the service user here.

Reply:
This is actually where we do want to use the request user; this is the call that's being made when users enter courses.

Reply:
Makes sense. I think we need to fix the [EnterpriseCustomerViewSet](https://github.com/edx/edx-enterprise/blob/master/enterprise/api/v1/views.py#L84) though and limit the GET results to those that the requesting user should have access to (i.e. the EnterpriseCustomer's that the user is linked to).

Reply:
Turns out we did need to use the service user here because there are flows through this code that do not have an authenticated request user.

Reply:
@zubair-arbi I think we need to completely refactor this function to use DB calls to look up the EnterpriseCustomer instead of API calls. We should create a `get_enterprise_customer(enterprise_customer_uuid)` function in this module to encapsulate those DB calls should we need to replace them with API calls somewhere down the line.